tests/cpp/test_gpu3.cpp

```diff
@@ -8296,7 +8296,52 @@ TEST_F(NVFuserTest, DecoupledDomains) {
       root_domain, logical_domain, allocation_domain, loop_domain, contiguity);
   TensorView* tv = IrBuilder::create<TensorView>(td, DataType::Float);
   auto all_ids = concat(logical_all, root_all, alloc_all, loop_all);
-  auto tv_all_vec = ir_utils::allIDsOf(tv);
+  auto tv_all_vec = tv->domain()->allIDs();
+  std::unordered_set<IterDomain*> tv_all(tv_all_vec.begin(), tv_all_vec.end());
+  EXPECT_EQ(tv_all, all_ids);
+}
+
+TEST_F(NVFuserTest, DecoupledDomains2) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto s0 = IrBuilder::create<Val>(DataType::Index);
+  auto s1 = IrBuilder::create<Val>(DataType::Index);
+  auto s2 = IrBuilder::create<Val>(DataType::Index);
+  auto s3 = IrBuilder::create<Val>(DataType::Index);
+  auto id0 = IterDomainBuilder(s0->fusion()->zeroVal(), s0).build();
+  auto id1 = IterDomainBuilder(s1->fusion()->zeroVal(), s1).build();
+  auto id2 = IterDomainBuilder(s2->fusion()->zeroVal(), s2).build();
+  auto id3 = IterDomainBuilder(s3->fusion()->zeroVal(), s3).build();
+  std::unordered_set<IterDomain*> all_ids{id0, id1, id2, id3};
+
+  AbstractTensor root_ids({id0, id1, id2, id3});
+
+  auto schedule = [&]() {
+    AbstractTensor domain = root_ids;
+    domain.merge(2);
+    all_ids.insert(domain[2].as<IterDomain*>());
+    domain.split(2, 256);
+    all_ids.insert(domain[2].as<IterDomain*>());
+    all_ids.insert(domain[3].as<IterDomain*>());
+    domain.merge(0);
+    all_ids.insert(domain[0].as<IterDomain*>());
+    domain.split(0, 256);
+    all_ids.insert(domain[0].as<IterDomain*>());
+    all_ids.insert(domain[1].as<IterDomain*>());
+    return domain.as<IterDomain*>();
+  };
+
+  auto logical_domain = schedule();
```

Comment:
Am I understanding correctly that this part just creates four different set of domains that all have the equivalent transformations? It would be great if you could add a short description on what behavior is tested.

Reply:
Yes, correct. Comment added.